### PR TITLE
improve sequence styling

### PR DESCRIPTION
**Pull request by Raemon**
– Fixes an issue where the description of a sequence had a blurry gradient
– Adds total wordcount to the large sequence items as well as the sequence hoverovers.
– Makes the image in large sequence items into a link.
– increase opacity for large sequence item images.

image

### CODE REVIEW

File: packages/lesswrong/components/sequences/LargeSequencesItem.tsx [hunk 1]
```diff
@@ -123,54 +134,61 @@ export const LargeSequencesItem = ({sequence, showAuthor=false, classes}: {
...
       <div className={classes.right}>
-        {sequence.chapters?.map((chapter) => <span key={chapter._id}>
-            {chapter.posts?.map(post => <SequencesSmallPostLink 
-                                          key={chapter._id + post._id} 
-                                          post={post}
-                                        />
+        {posts.map(post => <SequencesSmallPostLink 
+                              key={sequence._id + post._id} 
+                              post={post}
+                            />
```
Comment (jpaddison3) [inline packages/lesswrong/components/sequences/LargeSequencesItem.tsx:190]: Jim has [previously commented](https://github.com/ForumMagnum/ForumMagnum/pull/5105#discussion_r906367910) on this indentation style.

File: packages/lesswrong/components/sequences/SequencesHoverOver.tsx [hunk 2]
```diff
@@ -62,14 +72,15 @@ export const SequencesHoverOver = ({classes, sequence, showAuthor=true}: {
       />
     </ContentStyles>
     {!chapters && loading && <Loading />}
-    {chapters?.map((chapter) => <span key={chapter._id}>
-      {chapter.posts?.map(post => 
+    {posts.map(post => 
         <SequencesSmallPostLink
```
Comment (jpaddison3) [inline packages/lesswrong/components/sequences/SequencesHoverOver.tsx:76]: Now 4 indented

**Review by jpaddison3 — COMMENTED** [review]
The parts of this that affect the EA Forum seem fine. Made two minor comments on indentation. Haven't review the LW-only parts.

**Comment by Raemon** [comment]
Fixed indentation.

File: packages/lesswrong/components/sequences/LargeSequencesItem.tsx [hunk 3]
```diff
@@ -123,53 +134,60 @@ export const LargeSequencesItem = ({sequence, showAuthor=false, classes}: {
...
 }) => {
-  const { UsersName, ContentStyles, SequencesSmallPostLink, ContentItemTruncated } = Components
+  const { UsersName, ContentStyles, SequencesSmallPostLink, ContentItemTruncated, LWTooltip } = Components
   
   const [expanded, setExpanded] = useState<boolean>(false)
 
   const cloudinaryCloudName = cloudinaryCloudNameSetting.get()
 
+  let posts : PostsList[] = []
+  sequence.chapters?.forEach(chapter => chapter.posts?.forEach(post => posts.push(post)))
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/LargeSequencesItem.tsx:144]: can be `const posts = sequence.chapters?.flatMap(chapter => chapter.posts ?? [])`, I think?

File: packages/lesswrong/components/sequences/LargeSequencesItem.tsx [hunk 4]
```diff
@@ -123,53 +134,60 @@ export const LargeSequencesItem = ({sequence, showAuthor=false, classes}: {
...
+  const totalWordcount = posts.reduce((prev, curr) => prev + (curr?.contents?.wordCount || 0), 0)
+
+  const highlight = sequence.contents?.htmlHighlight || ""
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/LargeSequencesItem.tsx:147]: I prefer `??` to `||`, especially in cases where `0` or `''` are semantically meaningful values, to reduce confusion and possible sources of bugs.  (`??` will only coalesce for `null | undefined` on the left side, unlike `||`, which will also do it for `0`, `false`, `''`, etc.)

File: packages/lesswrong/components/sequences/LargeSequencesItem.tsx [hunk 5]
```diff
@@ -123,53 +134,60 @@ export const LargeSequencesItem = ({sequence, showAuthor=false, classes}: {
...
     <div className={classes.columns}>
       <div className={classes.left}>
-        <div className={classes.sequenceImage}>
-          <img className={classes.sequenceImageImg}
-            src={`https://res.cloudinary.com/${cloudinaryCloudName}/image/upload/c_fill,dpr_2.0,g_custom,h_96,q_auto,w_292/v1/${
-              sequence.gridImageId || "sequences/vnyzzznenju0hzdv6pqb.jpg"
-            }`}
-            />
-        </div>
+        <Link className={classes.imageLink} to={'/s/' + sequence._id}>
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/LargeSequencesItem.tsx:153]: nit: string template? 
```
`/s/${sequence._id}`
```

File: packages/lesswrong/components/sequences/LargeSequencesItem.tsx [hunk 6]
```diff
@@ -123,53 +134,60 @@ export const LargeSequencesItem = ({sequence, showAuthor=false, classes}: {
...
+              src={`https://res.cloudinary.com/${cloudinaryCloudName}/image/upload/c_fill,dpr_2.0,g_custom,h_96,q_auto,w_292/v1/${
+                sequence.gridImageId || "sequences/vnyzzznenju0hzdv6pqb.jpg"
+              }`}
+              />
+          </div>
+        </Link>
         <div className={classes.text}>
           <div className={classes.titleAndAuthor}>
-            <Link to={'/s/' + sequence._id} className={classes.title}>{sequence.title}</Link>
+            <Link  to={'/s/' + sequence._id} className={classes.title}>{sequence.title}</Link>
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/LargeSequencesItem.tsx:164]: nit: extra unnecessary space?

File: packages/lesswrong/components/sequences/SequencesHoverOver.tsx [hunk 7]
```diff
@@ -42,6 +48,10 @@ export const SequencesHoverOver = ({classes, sequence, showAuthor=true}: {
     fragmentName: 'ChaptersFragment',
     enableTotal: false,
   });
+
+  let posts : PostsList[] = []
+  chapters?.forEach(chapter => chapter.posts?.forEach(post => posts.push(post)))
+  const totalWordcount = posts.reduce((prev, curr) => prev + (curr?.contents?.wordCount || 0), 0)
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/SequencesHoverOver.tsx:54]: Same as:
https://github.com/ForumMagnum/ForumMagnum/pull/5181/files#r915343802
https://github.com/ForumMagnum/ForumMagnum/pull/5181/files#r915344286

**Review by darkruby501 — APPROVED** [review]
Approved subject to making requested changes

**Pull request merged**